openmdao/vectors/default_transfer.py

```diff
@@ -180,14 +181,73 @@ def _setup_discrete_transfers(group, recurse=True):
             Whether to call this method in subsystems.
         """
         group._discrete_transfers = transfers = defaultdict(list)
-        offset = len(group.pathname) + 1 if group.pathname else 0
+        name_offset = len(group.pathname) + 1 if group.pathname else 0
+
+        iproc = group.comm.rank
+        owns = group._owning_rank
 
         for tgt, src in iteritems(group._conn_discrete_in2out):
-            src_sys, src_var = src[offset:].split('.', 1)
-            tgt_sys, tgt_var = tgt[offset:].split('.', 1)
+            src_sys, src_var = src[name_offset:].split('.', 1)
+            tgt_sys, tgt_var = tgt[name_offset:].split('.', 1)
             xfer = (src_sys, src_var, tgt_sys, tgt_var)
             transfers[tgt_sys].append(xfer)
-            transfers[None].append(xfer)
```

Comment:
If group.comm.size == 1 then it looks like transfers[None] doesn't get defined.  I think we might need a new test as well since our current tests don't catch this.

Reply:
Put in story to capture this issue in the future